### PR TITLE
CMR-6365 modifying order of priority for sorting to check for usage

### PR DESCRIPTION
**Pull request by jaybarra**

### CODE REVIEW

File: search-app/src/cmr/search/data/elastic_relevancy_scoring.clj [hunk 1]
```diff
@@ -81,7 +91,7 @@
        ;; We only include this if one of the others is present
        (when (and (or use-temporal-sort? use-keyword-sort?)
```
Comment (ygliuvt) [inline search-app/src/cmr/search/data/elastic_relevancy_scoring.clj:92]: Add a `(not use-usage-sort?)` here to skip the unnecessary usage sort here.

File: search-app/src/cmr/search/data/elastic_relevancy_scoring.clj [hunk 2]
```diff
@@ -58,17 +58,27 @@
 
    The algorithm is to currently compare scores in the following order (we only go to the next level
    in the scoring system in the case of a tie at the higher level):
-   1. keyword boost
-   2. temporal overlap
-   3. community usage
-   4. temporal end date of the collection
-   5. processing level."
+   1. community usage
```
Comment (ygliuvt) [inline search-app/src/cmr/search/data/elastic_relevancy_scoring.clj:61]: nit: change this to `1. community usage, if usage_score sort key present`.

File: system-int-test/test/cmr/system_int_test/search/collection_sorting_search_test.clj [hunk 3]
```diff
@@ -504,4 +504,16 @@
...
+    (testing "sorting params"
+      (are [query-map items]
+           (d/refs-match-order? items
+                                (search/find-refs :collection
+                                                  query-map
+                                                  {:method :post}))
+
+           {:keyword "a*"} [c1 c6]
+           {:keyword "a*" :sort-key ["-usage_score"]} [c1 c6]
+           {:keyword "a*" :sort-key ["usage_score"]} [c6 c1]))))
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/collection_sorting_search_test.clj:518]: We should add a couple tests to show that `usage_score` sort key takes precedence over other sort_keys on keyword searches when multiple sort keys are present. e.g.
```
{:keyword "a*" :sort-key ["entry_title" "usage_score"]} [c6 c1]
{:keyword "a*" :sort-key ["-entry_title" "usage_score"]} [c6 c1]
{:keyword "a*" :sort-key ["entry_title" "-usage_score"]} [c1 c6]
{:keyword "a*" :sort-key ["-entry_title" "-usage_score"]} [c1 c6]
```

File: search-app/src/cmr/search/data/elastic_relevancy_scoring.clj [hunk 4]
```diff
@@ -58,17 +58,27 @@
...
    in the scoring system in the case of a tie at the higher level):
-   1. keyword boost
-   2. temporal overlap
-   3. community usage
-   4. temporal end date of the collection
-   5. processing level."
+   1. community usage, if usage_score sort key present
+   2. keyword boost
+   3. temporal overlap
+   4. relevancy with community usage secondary
```
Comment (sxu123) [inline search-app/src/cmr/search/data/elastic_relevancy_scoring.clj:64]: What does it mean exactly?  could you make it more clearly like "community usage, if ....  "  just like #1 above?
Also, could you correct the doc string of sort-use-relevancy-score in your ticket? there seem to be some typos there  which makes it difficult to read...

**Pull request merged**